src/IrishModel.cpp

```diff
@@ -10,6 +10,8 @@ IrishModel::IrishModel(const CommonModelInputParameters& commonParameters,
     fixedParameters_ = BuildFixedParameters(
         observations.waifw_norm.size(), commonParameters.paramlist
     );
+
+    Utilities::checkAndGetSize(observations.pf_pop, "observations.pf_pop");
```

Comment:
Notwithstanding my comment above, I'd suggest that once you have checked that the data are valid once in the IO code, there's no reason to check again in the model code.